src/clj/fluree/db/reasoner.cljc

```diff
@@ -218,24 +235,42 @@
             (recur r db**))
           db*)))))
 
+(defn deduplicate-raw-rules
```

Comment:
I think this could use a docstring to explain what it is doing. There is too much going on here and the code itself is not explaining it well.

In lieu of a docstring, perhaps if more of it was broken into multiple functions that were more descriptive.